ignite/contrib/handlers/lr_finder.py

```diff
@@ -256,12 +279,20 @@ def plot(
 
     def lr_suggestion(self) -> Any:
         """
-        Returns: learning rate at the minimum numerical gradient
+        Returns:
+            Learning rate at the minimum numerical gradient
+            (ignoring the increasing part of the curve)
         """
         if not self._history:
             raise RuntimeError("learning rate finder didn't run yet so lr_suggestion can't be returned")
         loss = self._history["loss"]
-        grads = torch.tensor([loss[i] - loss[i - 1] for i in range(1, len(loss))])
+        min_loss_idx = torch.tensor(loss).argmin()
+        # Ignore the increasing part of the curve
+        decreasing_losses = self._history["loss"][: int(min_loss_idx.item()) + 1]
+        if len(decreasing_losses) == 1:
+            raise RuntimeError("FastaiLRFinder got unexpected curve shape, the curve should be somehow U-shaped")
+        losses = torch.tensor(decreasing_losses)
+        grads = torch.tensor([losses[i] - losses[i - 1] for i in range(1, len(losses))])
```

Comment:
@KickItLikeShika finally, why we do not use more stable schema to determine grads with 3 points ?

Reply:
I tried to find a formula to find the grads with 3 points but I got nothing then, so I thought this way works really well so no need to change it, but I can dig more for the 3 points grads formula

Reply:
I had it already in my branch: `0.5 * (loss[i + 1] - loss[i - 1])` ?

Reply:
Thanks, I was getting it wrong! I will try to use it now and see the results

Reply:
3 points grads give very stable results while ignoring the increasing part, see here: https://colab.research.google.com/drive/1_siUuBjZkHVc_PGS28gmF0SYzKDgMbKJ?usp=sharing#scrollTo=sBpNO3Q9n55r
. I will update the `lr_suggestion` method now